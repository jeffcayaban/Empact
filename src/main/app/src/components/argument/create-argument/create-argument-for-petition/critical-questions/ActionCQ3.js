import React from 'react'
import { Panel } from "react-bootstrap";
import CreateArgumentOptions from "../../CreateArgumentOptions";
import {ARGUMENTATION_SCHEMES, CRITICAL_QUESTION_IDS} from "../../../../../constants";
import { showArgumentTemplate } from "../../../../../utils/Helpers";
import UserOpinionBtn from "../../../../common/UserOpinionBtn";
import GoalPreviewModal from "../../../../common/GoalPreviewModal";
import ActionPreviewModal from "../../../../common/ActionPreviewModal";

/**
 * Will display critical question "Does the action achieve the goal?" and provide options to create an argument in
 * response to the question.
 */

class ActionCQ3 extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            showOptions: false,
            showAction: false,
            showGoal: false,
            selectedArgScheme: ''
        };
    }

    handleShowArgumentOptions = () => this.setState({ showOptions: !this.state.showOptions });
    onOptionSelect = (optionKey) => this.setState({ selectedArgScheme: ARGUMENTATION_SCHEMES[optionKey]});

    render() {
        const { petition, petitionId, isSupporting } = this.props;
        const { showAction, showGoal, selectedArgScheme, showOptions } = this.state;

        console.log(isSupporting);

        const userSays = isSupporting ? 'does' : 'does not';
        const popularOpinionAssumption = `You believe it is generally well known that the proposed action ${userSays} achieve the goal.`;

        const actionArgumentTemplate = showArgumentTemplate(popularOpinionAssumption, selectedArgScheme,
            petitionId, isSupporting,isSupporting, null, CRITICAL_QUESTION_IDS[1], null);

        const actionPreviewBtnCallbackFn = () => this.setState({ showAction: true });
        const goalPreviewBtnCallbackFn = () => this.setState({ showGoal: true });

        return (
            <Panel expanded={showOptions} onToggle={()=>{}}>
                <Panel.Heading className={"argumentAssumption"}>
                    <div>
                        <div>
                            {`2. Can you provide a supporting argument as to why you think the `}
                            <button type="button" className="linkButton" onClick={actionPreviewBtnCallbackFn}>
                                action
                            </button>
                            {` ${userSays} achieve the `}
                            <button type="button" className="linkButton" onClick={goalPreviewBtnCallbackFn}>
                                goal
                            </button>
                            {`?`}
                        </div>
                        <ActionPreviewModal
                            petitionAction={petition.action}
                            showActionPreview={showAction}
                            container={this}
                            closeCallbackFn={() => this.setState({ showAction: false })}
                        />
                        <GoalPreviewModal
                            petitionGoal={petition.goal}
                            showGoalPreview={showGoal}
                            container={this}
                            closeCallbackFn={() => this.setState({ showGoal: false })}
                        />
                    </div>
                    <UserOpinionBtn showOption={showOptions} showArgTemplates={() => this.handleShowArgumentOptions()} />
                </Panel.Heading>
                <Panel.Collapse>
                    <Panel.Body>
                        <CreateArgumentOptions handleSelect={this.onOptionSelect.bind(this)} />
                        { actionArgumentTemplate }
                    </Panel.Body>
                </Panel.Collapse>
            </Panel>
        );
    }
}

export default ActionCQ3;
