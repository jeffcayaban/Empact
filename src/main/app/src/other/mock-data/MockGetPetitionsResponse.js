export default {
    "content": [
        {
            "id": 1,
            "title": "TestTitle",
            "situation": "TestSituation",
            "action": "TestAction",
            "goal": "TestGoal",
            "value": "TestValue",
            "creationDateTime": "2019-02-14T13:04:29Z",
            "lastUpdatedDateTime": "2019-02-14T13:04:38Z",
            "createdBy": {
                "id": 1,
                "username": "admin",
                "firstName": "Admin",
                "lastName": "Account",
                "authorities": [
                    "ROLE_ADMIN"
                ]
            },
            "closingDateTime": "2019-02-15T13:04:27Z",
            "isAnonymous": false,
            "isWinning": true
        }
    ],
    "page": 0,
    "size": 5,
    "totalElements": 1,
    "totalPages": 1,
    "last": true
}