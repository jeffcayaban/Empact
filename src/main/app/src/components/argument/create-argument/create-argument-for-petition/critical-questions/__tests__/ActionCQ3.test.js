import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import ActionCQ3 from "../ActionCQ3";

import MockGetPetitionsResponse from '../../../../../../other/mock-data/MockGetArgumentsResponse';

describe('ActionCQ3', () => {

    it('should render correctly', () => {
        const petitionData = MockGetPetitionsResponse.content[0];

        const component = shallow(
            <ActionCQ3
                petition={petitionData}
                isSupporting={true}
                petitionId={petitionData.id}
            />
        );

        expect(toJson(component)).toMatchSnapshot();
    });

});
