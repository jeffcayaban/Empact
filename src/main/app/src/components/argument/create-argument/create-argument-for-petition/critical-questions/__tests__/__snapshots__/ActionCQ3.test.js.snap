// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCQ3 should render correctly 1`] = `
<Uncontrolled(Panel)
  expanded={false}
  onToggle={[Function]}
>
  <PanelHeading
    bsClass="panel"
    className="argumentAssumption"
    componentClass="div"
  >
    <div>
      <div>
        2. Can you provide a supporting argument as to why you think the 
        <button
          className="linkButton"
          onClick={[Function]}
          type="button"
        >
          action
        </button>
         does achieve the 
        <button
          className="linkButton"
          onClick={[Function]}
          type="button"
        >
          goal
        </button>
        ?
      </div>
      <ActionPreviewModal
        closeCallbackFn={[Function]}
        container={
          ActionCQ3 {
            "context": Object {},
            "handleShowArgumentOptions": [Function],
            "onOptionSelect": [Function],
            "props": Object {
              "isSupporting": true,
              "petition": Object {
                "agreesWithCQ": true,
                "argumentType": "PopularOpinion",
                "createdBy": Object {
                  "authorities": Array [
                    "ROLE_ADMIN",
                  ],
                  "firstName": "Admin",
                  "id": 1,
                  "lastName": "Account",
                  "username": "admin",
                },
                "creationDateTime": "2019-02-14T13:04:38Z",
                "criticalQuestionId": "CQ2",
                "explanation": "TestExplanation",
                "id": 1,
                "isAnonymous": false,
                "isSupporting": true,
                "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
                "parentArgumentId": null,
                "rootPetitionId": 1,
                "sources": Array [],
              },
              "petitionId": 1,
            },
            "refs": Object {},
            "setState": [Function],
            "state": Object {
              "selectedArgScheme": "",
              "showAction": false,
              "showGoal": false,
              "showOptions": false,
            },
            "updater": Updater {
              "_callbacks": Array [],
              "_renderer": ReactShallowRenderer {
                "_context": Object {},
                "_currentlyRenderingComponent": null,
                "_didScheduleRenderPhaseUpdate": false,
                "_dispatcher": Object {
                  "readContext": [Function],
                  "useCallback": [Function],
                  "useContext": [Function],
                  "useDebugValue": [Function],
                  "useEffect": [Function],
                  "useImperativeHandle": [Function],
                  "useLayoutEffect": [Function],
                  "useMemo": [Function],
                  "useReducer": [Function],
                  "useRef": [Function],
                  "useState": [Function],
                },
                "_element": <ActionCQ3
                  isSupporting={true}
                  petition={
                    Object {
                      "agreesWithCQ": true,
                      "argumentType": "PopularOpinion",
                      "createdBy": Object {
                        "authorities": Array [
                          "ROLE_ADMIN",
                        ],
                        "firstName": "Admin",
                        "id": 1,
                        "lastName": "Account",
                        "username": "admin",
                      },
                      "creationDateTime": "2019-02-14T13:04:38Z",
                      "criticalQuestionId": "CQ2",
                      "explanation": "TestExplanation",
                      "id": 1,
                      "isAnonymous": false,
                      "isSupporting": true,
                      "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
                      "parentArgumentId": null,
                      "rootPetitionId": 1,
                      "sources": Array [],
                    }
                  }
                  petitionId={1}
                />,
                "_firstWorkInProgressHook": null,
                "_forcedUpdate": false,
                "_instance": [Circular],
                "_isReRender": false,
                "_newState": null,
                "_numberOfReRenders": 0,
                "_previousComponentIdentity": null,
                "_renderPhaseUpdates": null,
                "_rendered": <Uncontrolled(Panel)
                  expanded={false}
                  onToggle={[Function]}
                >
                  <PanelHeading
                    bsClass="panel"
                    className="argumentAssumption"
                    componentClass="div"
                  >
                    <div>
                      <div>
                        2. Can you provide a supporting argument as to why you think the 
                        <button
                          className="linkButton"
                          onClick={[Function]}
                          type="button"
                        >
                          action
                        </button>
                         does achieve the 
                        <button
                          className="linkButton"
                          onClick={[Function]}
                          type="button"
                        >
                          goal
                        </button>
                        ?
                      </div>
                      <ActionPreviewModal
                        closeCallbackFn={[Function]}
                        container={[Circular]}
                        showActionPreview={false}
                      />
                      <GoalPreviewModal
                        closeCallbackFn={[Function]}
                        container={[Circular]}
                        showGoalPreview={false}
                      />
                    </div>
                    <UserOpinionBtn
                      showArgTemplates={[Function]}
                      showOption={false}
                    />
                  </PanelHeading>
                  <PanelCollapse
                    bsClass="panel"
                  >
                    <PanelBody
                      bsClass="panel"
                      collapsible={false}
                    >
                      <CreateArgumentOptions
                        handleSelect={[Function]}
                      />
                    </PanelBody>
                  </PanelCollapse>
                </Uncontrolled(Panel)>,
                "_rendering": false,
                "_updater": [Circular],
                "_workInProgressHook": null,
              },
            },
            Symbol(enzyme.__setState__): [Function],
          }
        }
        showActionPreview={false}
      />
      <GoalPreviewModal
        closeCallbackFn={[Function]}
        container={
          ActionCQ3 {
            "context": Object {},
            "handleShowArgumentOptions": [Function],
            "onOptionSelect": [Function],
            "props": Object {
              "isSupporting": true,
              "petition": Object {
                "agreesWithCQ": true,
                "argumentType": "PopularOpinion",
                "createdBy": Object {
                  "authorities": Array [
                    "ROLE_ADMIN",
                  ],
                  "firstName": "Admin",
                  "id": 1,
                  "lastName": "Account",
                  "username": "admin",
                },
                "creationDateTime": "2019-02-14T13:04:38Z",
                "criticalQuestionId": "CQ2",
                "explanation": "TestExplanation",
                "id": 1,
                "isAnonymous": false,
                "isSupporting": true,
                "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
                "parentArgumentId": null,
                "rootPetitionId": 1,
                "sources": Array [],
              },
              "petitionId": 1,
            },
            "refs": Object {},
            "setState": [Function],
            "state": Object {
              "selectedArgScheme": "",
              "showAction": false,
              "showGoal": false,
              "showOptions": false,
            },
            "updater": Updater {
              "_callbacks": Array [],
              "_renderer": ReactShallowRenderer {
                "_context": Object {},
                "_currentlyRenderingComponent": null,
                "_didScheduleRenderPhaseUpdate": false,
                "_dispatcher": Object {
                  "readContext": [Function],
                  "useCallback": [Function],
                  "useContext": [Function],
                  "useDebugValue": [Function],
                  "useEffect": [Function],
                  "useImperativeHandle": [Function],
                  "useLayoutEffect": [Function],
                  "useMemo": [Function],
                  "useReducer": [Function],
                  "useRef": [Function],
                  "useState": [Function],
                },
                "_element": <ActionCQ3
                  isSupporting={true}
                  petition={
                    Object {
                      "agreesWithCQ": true,
                      "argumentType": "PopularOpinion",
                      "createdBy": Object {
                        "authorities": Array [
                          "ROLE_ADMIN",
                        ],
                        "firstName": "Admin",
                        "id": 1,
                        "lastName": "Account",
                        "username": "admin",
                      },
                      "creationDateTime": "2019-02-14T13:04:38Z",
                      "criticalQuestionId": "CQ2",
                      "explanation": "TestExplanation",
                      "id": 1,
                      "isAnonymous": false,
                      "isSupporting": true,
                      "lastUpdatedDateTime": "2019-02-14T13:04:51Z",
                      "parentArgumentId": null,
                      "rootPetitionId": 1,
                      "sources": Array [],
                    }
                  }
                  petitionId={1}
                />,
                "_firstWorkInProgressHook": null,
                "_forcedUpdate": false,
                "_instance": [Circular],
                "_isReRender": false,
                "_newState": null,
                "_numberOfReRenders": 0,
                "_previousComponentIdentity": null,
                "_renderPhaseUpdates": null,
                "_rendered": <Uncontrolled(Panel)
                  expanded={false}
                  onToggle={[Function]}
                >
                  <PanelHeading
                    bsClass="panel"
                    className="argumentAssumption"
                    componentClass="div"
                  >
                    <div>
                      <div>
                        2. Can you provide a supporting argument as to why you think the 
                        <button
                          className="linkButton"
                          onClick={[Function]}
                          type="button"
                        >
                          action
                        </button>
                         does achieve the 
                        <button
                          className="linkButton"
                          onClick={[Function]}
                          type="button"
                        >
                          goal
                        </button>
                        ?
                      </div>
                      <ActionPreviewModal
                        closeCallbackFn={[Function]}
                        container={[Circular]}
                        showActionPreview={false}
                      />
                      <GoalPreviewModal
                        closeCallbackFn={[Function]}
                        container={[Circular]}
                        showGoalPreview={false}
                      />
                    </div>
                    <UserOpinionBtn
                      showArgTemplates={[Function]}
                      showOption={false}
                    />
                  </PanelHeading>
                  <PanelCollapse
                    bsClass="panel"
                  >
                    <PanelBody
                      bsClass="panel"
                      collapsible={false}
                    >
                      <CreateArgumentOptions
                        handleSelect={[Function]}
                      />
                    </PanelBody>
                  </PanelCollapse>
                </Uncontrolled(Panel)>,
                "_rendering": false,
                "_updater": [Circular],
                "_workInProgressHook": null,
              },
            },
            Symbol(enzyme.__setState__): [Function],
          }
        }
        showGoalPreview={false}
      />
    </div>
    <UserOpinionBtn
      showArgTemplates={[Function]}
      showOption={false}
    />
  </PanelHeading>
  <PanelCollapse
    bsClass="panel"
  >
    <PanelBody
      bsClass="panel"
      collapsible={false}
    >
      <CreateArgumentOptions
        handleSelect={[Function]}
      />
    </PanelBody>
  </PanelCollapse>
</Uncontrolled(Panel)>
`;
